var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -113,6 +113,14 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
 
     # https://gcc.gnu.org/install/prerequisites.html
     depends_on('gmp@4.3.2:')
+    # mawk is not sufficient for go support
+    depends_on('gawk@3.1.5:')
+    depends_on('texinfo@4.7:', type='build')
+    # dependencies required for git versions
+    depends_on('m4@1.4.6:', type='build')
+    depends_on('automake@1.15.1:', type='build')
+    depends_on('autoconf@2.69:', type='build')
```

Comment:
```suggestion
    depends_on('m4@1.4.6:', when='@master', type='build')
    depends_on('automake@1.15.1:', when='@master', type='build')
    depends_on('autoconf@2.69:', when='@master', type='build')
```

Reply:
I intentionally did not do that. It is almost correct, but means that they cannot apply for git-versioned builds. Lacking a way to apply to all such builds this results in more working configurations than limiting them to master.

Reply:
By "git-versioned builds" you mean that thing where Spack lets you use a git commit to specify which version of a package to build? Hmm, that seems like a fundamental flaw. I don't think we want to add this dependency for all users when 99% of people simply use the latest stable release (the default version).

Reply:
It is most definitely a fundamental flaw, it's nearly impossible for many packages to actually make them work sensibly, but not to block them. I suppose you have a point about releases versions, it doesn't hurt too much to limit them to the one listed development version.